### PR TITLE
Implement ucsCallback job

### PR DESCRIPTION
**Pull request by bbcyyb**
### Background
UCSM http response is slower than RackHD in some cases and will lead to RackHD fails to get data from UCSM via HTTP. Thus in ucs-service we implemented asynchronous API to avoid this and data will be post back to RackHD via a callback API.

### Details
Add new base job for ucs related job.
Unit test is included.

Depends on https://github.com/RackHD/on-http/pull/802, https://github.com/RackHD/ucs-service/pull/28

Paired with: @pengz1
@anhou @pengz1 @mcgG

### CODE REVIEW

File: lib/jobs/ucs-base-job.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            .then(function(res) {
+                if (res && res.body && res.body.toUpperCase() !== "ACCEPTED") {
+                    throw new Error(
+                        "Request was not ACCEPTED. Please check input parameters.");
+                }
+                return self._subscribeHttpResponseUuidByPromisify(taskId);
+            });
+    };
+
+    UcsBaseJob.prototype._subscribeHttpResponseUuidByPromisify = function(id) {
```
Comment (anhou) [inline lib/jobs/ucs-base-job.js]: Good job!

Comment (lanchongyizu) [reply lib/jobs/ucs-base-job.js]: `_subscribeHttpResponseUuid `is already a `promise`(https://github.com/RackHD/on-tasks/blob/master/lib/jobs/base-job.js#L230), we don't need to promisify it again.

Comment (lanchongyizu) [reply lib/jobs/ucs-base-job.js]: OK, since you want to Repromisify `_subscribeHttpResponseUuid`, errors should be processed by calling `callback`.

Comment (bbcyyb) [reply lib/jobs/ucs-base-job.js]: Yes, current code will eat all error scenes. I'm going to add error handler for asynchronous processing, to make sure they can be found and handled.

**Comment by bbcyyb** [comment]
Change parameter from taskId to a uuid string named callbackId which is used to define a amqp channel.

**Pull request merged**